src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -142,7 +147,10 @@ public String createJwt(
             throw new Exception("Roles cannot be null");
         }
 
-        /* TODO: If the backendRoles is not null and the BWC Mode is on, put them into the "dbr" claim */
+        if (bwcModeEnabled && backendRoles != null) {
```

Comment:
@DarshitChanpura  i wonder if just keeping the format used a few lines above would be ok, or if for the backwards compatibility to work we should keep the exact format currently used when adding to the Context:

`            StringJoiner joiner = new StringJoiner("|");
            joiner.add(user.getName());
            joiner.add(String.join(",", user.getRoles()));
            joiner.add(String.join(",", Sets.union(user.getSecurityRoles(), mappedRoles)));`

Reply:
I think the idea behind this is to send backend roles unencrypted if its in backwards compatibility mode. Can you please elaborate on `keep the exact same format`?

Reply:
The code snipped above that i pasted in the above post is how the back end roles are added to the context (joinined with " | " symbol and  adding the user's name  at the begininng ). 

My question if we should populate using the same format as the plugins might be expecting/parsing that exact format and could fail if the joiner character is different. 

(Im not aware if this is the case or not)

Reply:
Ah I see, this is only for token parsing so we should be good with the format you have in this PR. @RyanL1997 Can you confirm?